ch32fun/ch57xhw.h

```diff
@@ -1628,6 +1633,172 @@ typedef enum
 #define R8_UEP7_T_LEN       (*((vu8*)0x4000806C))  // endpoint 7 transmittal length
 #define R8_UEP7_CTRL        (*((vu8*)0x4000806E))  // endpoint 7 control
 
+//some probabily works for ch571/3 too, but not tested
+//so only for ch570/2 for now.
+#if( MCU_PACKAGE == 2 || MCU_PACKAGE == 0) // CH570/2
+
+#define RTC_MAX_COUNT       0xA8BFFFFF
+
+// LSI frequency is actually unknown and can't be tuned. 
+// It can be anywhere between 24 kHz to 42 kHz. If percise timing is needed,
+// the user should calibrate it with R8_OSC_CAL_CTRL and
+// R16_OSC_CAL_CNT.
+#define RTC_FREQ            32000 
+
+// The datasheet of ch570/2 refer to some registers differently.
+// They should be defined in case the user is referring to the datasheet.
+#define R16_RTC_CNT_LSI     R16_RTC_CNT_32K 
+#define R16_RTC_CNT_DIV1    R16_RTC_CNT_2S 
+#define R8_LSI_CONFIG       R8_CK32K_CONFIG
+#define RB_CLK_LSI_PON      RB_CLK_XT32K_PON
+
+// Additional power plan bits.
+#define RB_PWR_LDO5V_EN     0x0100
+#define RB_PWR_RAM12K       RB_PWR_RAM2K
+#define RB_PWR_RAM24K       RB_PWR_RAM12K
+#define RB_XT_PRE_EN        0 //no such bit for ch570/2
+
+
+#define CLK_PER_US          (1.0 / ((1.0 / RTC_FREQ) * 1000 * 1000))
+#define CLK_PER_MS          (CLK_PER_US * 1000)
+#define US_TO_RTC(us)       ((uint32_t)((us) * CLK_PER_US + 0.5))
+#define MS_TO_RTC(ms)       ((uint32_t)((ms) * CLK_PER_MS + 0.5))
+
+#define SLEEP_RTC_MIN_TIME  US_TO_RTC(1000)
+#define SLEEP_RTC_MAX_TIME  (RTC_MAX_COUNT - 1000 * 1000 * 30)
+#define WAKE_UP_RTC_MAX_TIME US_TO_RTC(1600)
+
+RV_STATIC_INLINE void PFIC_EnableIRQ(IRQn_Type IRQn)
```

Comment:
this is only used once, please remove this in favor of the comment I made where it is used.